flarefly/fitter.py

```diff
@@ -702,27 +737,35 @@ def get_chi2(self):
         chi2: float
             chi2
         """
-        # for chi2 loss, just retrieve loss value in fit result
-        if self._chi2_loss_:
-            chi2 = float(self._fit_result_.loss.value())
-        # for nll loss, compute chi2 "by hand"
-        else:
-            chi2 = 0
+
+        chi2 = 0
+        norm = self._data_handler_.get_norm()
+        if self._data_handler_.get_is_binned():
+            # for chi2 loss, just retrieve loss value in fit result
+            if self._chi2_loss_:
+                return float(self._fit_result_.loss.value())
+
+            # for nll loss, compute chi2 "by hand"
             # access normalized data values and errors for all bins
             binned_data = self._data_handler_.get_binned_data()
-            norm = self._data_handler_.get_norm()
-            data_values = binned_data.values()/norm
-            data_variances = binned_data.variances()/norm/norm
+            data_values = binned_data.values()
+            data_variances = binned_data.variances()
             # access model predicted values
-            model_values = self._total_pdf_binned_.values()
+            model_values = self._total_pdf_binned_.values()*norm
             # compute chi2
-            bins = len(data_values)
-            for ibin in range(bins):
-                residual = data_values[ibin] - model_values[ibin]
-                denom = data_variances[ibin]
-                chi2 += residual*residual/denom
+            for (data, model, data_variance) in zip(data_values, model_values, data_variances):
+                chi2 += (data - model)*(data - model)/data_variance
+            return chi2
 
-        return chi2
+        # for unbinned data
+        data_values = self._data_handler_.get_binned_data_from_unbinned_data()
+        # access model predicted values
+        model_values = self._total_pdf_binned_.values()*norm
+        # compute chi2
+        for (data, model) in zip(data_values, model_values):
+            chi2 += (data - model)*(data - model)/model
```

Comment:
```suggestion
            chi2 += (data - model)**2/model
```
Shouldn't the denominator be the error on the data?